### ISSUE TITLE
01. Dataset batch に source / receiver / trace metadata を出す

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#398]
# 01. Dataset batch に source / receiver / trace metadata を出す

## 目的

MAE / BlindTrace で得た pilot trace と observed trace の lag を、後段の `seis-statics` に渡せるようにするため、seisai の dataset / dataloader batch に物理 trace metadata を含める。

MVP では solver は `source_id` と `receiver_id` のみを使用する。ただし、後続比較で CMP / offset / azimuth / view-bias を追加できるよう、可能な metadata は同時に保持する。

## 背景

MVP の静補正モデルは以下に限定する。

```text
lag_i ≈ S_source(i) + R_receiver(i)
```

したがって、各 trace について最低限以下が必要になる。

```text
trace_id
source_id
receiver_id
```

`seis-statics` 側は MAE / denoising / SEG-Y I/O を持たないため、これらの metadata 生成は seisai 側の責務とする。

## 編集対象候補

- `packages/seisai-dataset/src/seisai_dataset/file_info.py`
- `packages/seisai-dataset/src/seisai_dataset/segy_gather_base.py`
- `packages/seisai-dataset/src/seisai_dataset/segy_gather_pipeline_dataset.py`
- `packages/seisai-dataset/src/seisai_dataset/sample_flow.py`
- 必要に応じて関連 tests

## 実装内容

1. `FileInfo` から subset trace に対応する header arrays を取り出せる helper を追加する。
2. `BaseSegyGatherPipelineDataset._build_output_extras_common` または pipeline 固有の `_build_output_extras` で、以下を batch output に含める。

   ```text
   trace_indices          int64, shape (H,)
   trace_id               int64 or str-compatible, shape (H,)
   source_id              int64, shape (H,)
   receiver_id            int64, shape (H,)
   file_index             int64 scalar or shape (H,)
   file_path              existing
   dt_sec / sample_interval_s
   trace_valid            existing if available
   ```

3. 後続拡張用として、存在する場合は以下も出す。

   ```text
   cmp_id                 int64, shape (H,)
   offset                 float32, shape (H,)
   source_x_m             float32, shape (H,) optional
   source_y_m             float32, shape (H,) optional
   receiver_x_m           float32, shape (H,) optional
   receiver_y_m           float32, shape (H,) optional
   geometry_valid_mask    bool, shape (H,) optional
   ```

4. `receiver_id` の初期実装は既存 `chno_values` でよい。ただし、将来 geometry から receiver station key を作れるよう、命名とコメントで `receiver_id` と `chno` を分ける。
5. padding / superwindow / transform 後の trace order と metadata order が必ず一致するようにする。
6. batch collation 後も torch Tensor として保持されることを確認する。

## 受け入れ条件

- BlindTrace training / inference 用 dataloader batch に `source_id` と `receiver_id` が含まれる。
- `source_id.shape == receiver_id.shape == trace_indices.shape == (H,)` である。
- `trace_valid` がある場合、padding trace について metadata が安全に扱える。
- 既存の BlindTrace training test が壊れない。
- `primary_keys=[ffid]`, `primary_keys=[cmp]` など既存 sampler 設定で動作する。

## テスト

- 小さな synthetic / test SEG-Y で dataset sample を1件取り、metadata shape と dtype を検証する。
- `trace_indices` で `FileInfo.ffid_values`, `chno_values`, `cmp_values`, `offsets` を参照した値と、batch metadata が一致することを検証する。
- `trace_valid` がある場合、valid trace だけを比較対象にする。

## 非対象

- `seis-statics` の import。
- NCC lag 推定。
- source/receiver statics solve。
- CMP / offset term の solver 実装。
